pandas/io/excel/_base.py

```diff
@@ -513,6 +515,11 @@ def parse(
                 else:
                     offset = 1 + max(header)
 
+                # GH34673: if MultiIndex names present and not defined in the header,
+                # don't forward fill the names
+                if has_index_names:
+                    offset += 1
```

Comment:
I could be wrong but the +1 feels a little arbitrary here - what would happen if there were say two consecutive rows in the column with missing data?

Reply:
Good point, thanks for bringing this up. Will add another test for this, but I think the +1 is ok here. It's not about handling 1 empty row, the problem was that `offset` was the line containing MI names when index names were present. `offset` is the position from which forward filling logic started, so the names were being used to forward fill missing MI values. The +1 is to make sure that `offset` points to the first line of MI values instead, so forward filling only includes actual values.

Reply:
This explanation (paraphrased) would be good to include as a comment.

Reply:
Added test for multiple blanks after name and a clearer comment